web/src/stream_events.ts

```diff
@@ -28,20 +28,47 @@ import * as stream_muting from "./stream_muting.ts";
 import * as stream_settings_api from "./stream_settings_api.ts";
 import * as stream_settings_ui from "./stream_settings_ui.ts";
 import * as sub_store from "./sub_store.ts";
+import type {StreamSubscription} from "./sub_store.ts";
 import * as unread_ui from "./unread_ui.ts";
 import * as user_profile from "./user_profile.ts";
 
 // In theory, this function should apply the account-level defaults,
 // however, they are only called after a manual override, so
 // doing so is unnecessary with the current code.  Ideally, we'd do a
 // refactor to address that, however.
-function update_stream_setting(sub, value, setting) {
-    const $setting_checkbox = $(`#${CSS.escape(setting)}_${CSS.escape(sub.stream_id)}`);
+function update_stream_setting(
+    sub: StreamSubscription,
+    value: boolean | null,
+    setting:
+        | "desktop_notifications"
+        | "audible_notifications"
+        | "push_notifications"
+        | "email_notifications"
+        | "wildcard_mentions_notify"
+        | "pin_to_top",
+): void {
+    const $setting_checkbox = $(`#${CSS.escape(setting)}_${CSS.escape(sub.stream_id.toString())}`);
     $setting_checkbox.prop("checked", value);
+    if (setting === "pin_to_top") {
+        assert(value !== null);
+        sub[setting] = value;
+        return;
+    }
```

Comment:
This `if` statement looks new? But I guess necessary to type-narrow properly.

Reply:
Exactly, added for typing reasons.